### PR TITLE
Graph version update

### PR DESCRIPTION
**Pull request by shuagarw**
The content to be added to Changelog is as follows:
* Added public Graph client 'azure-graph'
* Upgraded graph api-version to 1.6-internal
* Fix breaking changes in graph commands

### CODE REVIEW

File: lib/util/utils.js [hunk 1]
```diff
@@ -73,14 +73,37 @@ exports.createClient = function (factoryMethod, credentials, endpoint, options)
...
 
-exports.createAutoRestClient = function (FactoryMethod, subscription) {
+/**
+ * Create Autorest client
+ *
+ * @param {function} Method to create an instance of the client
+ * 
+ * @param {object} Subscription object
+ * 
+ * @param {object} [options] checks for isTenantBased and endpoint properties
```
Comment (amarzavery) [inline lib/util/utils.js]:
```
* @param {object} [options] - Optional parameters
* @param {bool} [options.isTenantBased] - A Boolean value (true) indicating that the client is tenant based
* @param {string} [options.endPoint] - The url specifying the endpoint (baseUrl) for the client
```

File: lib/util/utils.js [hunk 2]
```diff
@@ -73,14 +73,37 @@ exports.createClient = function (factoryMethod, credentials, endpoint, options)
...
-  var subscriptionId = subscription.id;
-  var baseUri = exports.stringTrimEnd(subscription.resourceManagerEndpointUrl, '/');
+  var credentials = null;
+  var baseUri = null;
+  var subscriptionOrTenantId = null;
+
+  if (options && options.isTenantBased && options.isTenantBased === true && options.endPoint) {
+    subscriptionOrTenantId = subscription.tenantId;
+    credentials = subscription._createCredentials(options.endPoint);
+    baseUri = exports.stringTrimEnd(subscription.activeDirectoryGraphResourceId, '/');
```
Comment (amarzavery) [inline lib/util/utils.js]: baseUri = exports.stringTrimEnd(options.endpoint, '/');

File: test/commands/arm/ad/arm.ad-tests.js [hunk 3]
```diff
@@ -162,11 +162,12 @@ describe('arm', function () {
         });
       });
 
-      it('should work to list and show users', function (done) {
+      it.skip('should work to list and show users', function (done) {
```
Comment (amarzavery) [inline test/commands/arm/ad/arm.ad-tests.js]: The ad tests should pass, correct? Why is this skipped?

File: lib/util/utils.js (hunk 2 above)
Comment (amarzavery) [inline lib/util/utils.js]: We should not couple having an endpoint with isTenantBased.
if options.endPoint is provided then use it, else we default to resourceManagerEndpointUrl

File: lib/util/utils.js [hunk 4]
```diff
@@ -73,14 +73,37 @@ exports.createClient = function (factoryMethod, credentials, endpoint, options)
...
+ * @returns {object} An instance of the client
+ */
+
+exports.createAutoRestClient = function (FactoryMethod, subscription, options) {
   ensureAADBackedSubscription(subscription);
-  var credentials = subscription._createCredentials();
-  var subscriptionId = subscription.id;
-  var baseUri = exports.stringTrimEnd(subscription.resourceManagerEndpointUrl, '/');
+  var credentials = null;
+  var baseUri = null;
```
Comment (amarzavery) [inline lib/util/utils.js]: initialize baseUri with resourceManagerEndpointUrl

Comment (shuagarw) [inline lib/util/utils.js]: Fixed checked in.

File: test/commands/arm/ad/arm.ad-tests.js (hunk 3 above)
Comment (shuagarw) [inline test/commands/arm/ad/arm.ad-tests.js]: As we discussed this is due to a bug in autorest generated code. Will wait for fix to be in to unskip this.

**Pull request merged**